docs/add-ons/presentation.md

```diff
@@ -83,3 +107,55 @@ Ingress API gateway supports the following:
 - HTTP/1.x
 - Streaming content
 - Websockets
+
+Ingress grants your Add-on 2 points of security and can be implemented easily from bash as a separate, no-dependencies, background process with the default home-assistant image, as follows. Please note, this is a basic static webserver setup and provides little benefit over the log tab. Additionally, this implementation has less security than any standard HTTP Framework, but it is intended to run within a controlled environment between the shell script and the Ingress server in Home Assistant, and is not exposed directly to the Internet or any network. 
+
+Place this line somewhere near the top of your run.sh.  This will start a `nc` listening server and run the `/opt/server.sh` each time a user connects to your Ingress. The output from stdout will be displayed as a response from the server. 
+``` bash
+nc -lk -p 8099 -e  exec /opt/server.sh 3>/dev/null &
+```
+Create a `rootfs/opt` folder within your Add-on folder. Then create a `server.sh` file in that folder. The server.sh will output the HTTP Headers and HTML required to display information.
+
+/opt/server.sh
+```bash
+#HTTP Server Headers
+echo -e 'HTTP/1.1 200 OK\r\nServer: DeskPiPro\r\nDate:$(date)\r\nContent-Type: text/html; charset=UTF8\r\nCache-Control: no-store, no cache, must-revalidate\r\n\r\n'
```

Comment:
There is really no general use in implementing this in bash. It makes this example unclear and unpractical.

Ingress is meant to pull out the application that is running inside the add-on into the Home Assistant frontend. Not gaining points with a bash script.

Reply:
Bash is the lowest level which can achieve the stated goal.  Note the use of a log.html file within the server.sh file which is intended to pull information from the application running inside the add-on. 

Additionally, 
- This methodology utilizes 100% built-in methods provided by Home Assistant's busybox and bash. 
- Low power consumption and footprint contribute to fast/lightweight installation
- This adds almost 0 cpu overhead from external binaries such as Python and associated libraries.  
- The above are all important considerations on low power hardware such as RPi and embedded systems in general.

I'm using this technique to display more information than is possible within the log and in a more clear format within the [Run On Startup.d add-on](https://community.home-assistant.io/t/run-on-startup-d/271008).  The alternative would leave my users log-less and reduce the UX, as the built-in logs only scroll so high and lack rich text capability.

Reply:
How can we make this example more clear?

Reply:
> Bash is the lowest level that can achieve the stated goal.

That is true, however, it is not a goal common goal for using Ingress.

The goal is to expose an application interface.

Reply:
I understand that you feel that this is not the way you intended add-ons to implement Ingress.  However, consider the alternative.  Any other example would require recommending additional packages be installed and explaining how they work.  This is the only method which does not require recommending installation of Python, Java, Node, Apache HTTPD, or other heavy-weight package.  This is the rawest form of Ingress we can possibly recommend and gets the users up and running quickly when developing a package intended exclusively for Home Assistant.  

When dealing with repackaging an existing app, setting port 8099 is easy. When bringing up an app exclusively for the Home Assistant environment, this example provides a way to perform mockups and establishes communication within the HA environment.

Reply:
> The goal is to expose an application interface.
I understand that this is not the intended method of utilization.  However, any other method of explanation requires external dependencies and explanation of those external dependencies.  Eg. setting up a server in Python, NodeJS, Java, Apache HTTPD, or other.  This requires 0-explanation of dependencies and gives the user a method to see immediate results without any additional overhead and because of this, is perfect for a clean example, creating app mockups, and quick-start into the world of Ingress.

Admittedly, as a security engineer, that 2-extra points of security drives me :). Undoubtedly that has an effect on my decisions on how to create my own add-ons.  I feel it should motivate any experienced developer as well. 

I suppose the real question here is how can we explain an example of ingress without recommending a platform, explaining development, going into APIs, or getting too detailed in our example?  Is there another way within HA?  Is there some Hello World of web servers?

Reply:
> is perfect for a clean example

I'm sorry, in this, we are on totally different pages. It sets an example that is not relevant. And I'm really not going to approve this example for merging.

A similar example could have been given by installing a simple nginx, which has a default page it shows.

Reply:
Will work on it over the weekend.  You're the boss.

Reply:
I submitted an Nginx example, but I'm curious as to how you recommend submitting this extra information cleanly.  
---
Since no project is complete without AppArmor, this is an example [apparmor.txt](https://pastebin.adamoutler.com/eZTdgae1uQiH9tmV).  It may be applied to improve security and reduce security risks. Tailoring an apparmor to Ingress can be challenging, but the logs will generally explain any problems encountered.